### ISSUE TITLE
SQL_Latin1_General_CP1_CS_AS is not generate a valid code to Update-Database

### ISSUE THREAD

**Issue by Ali-YousefiTelori** [dotnet/efcore#24820]
When I try to add 

```csharp
modelBuilder.Entity(entity)Property(x=>x.UniqueRecordId).HasColumnType("NVARCHAR(4000) COLLATE SQL_Latin1_General_CP1_CS_AS")
```

 to my model builder my migration will add this line:

```csharp
migrationBuilder.CreateTable(
                name: "ObjectIds",
                columns: table => new
                {
                    Id = table.Column<long>(type: "bigint", nullable: false)
                        .Annotation("SqlServer:Identity", "1, 1"),
                    ObjectId = table.Column<string>(type: "nvarchar(max)", nullable: true),
                    ModificationDateTime = table.Column<DateTime>(type: "datetime2", nullable: false),
                    CreationDateTime = table.Column<DateTime>(type: "datetime2", nullable: false),
                    Key = table.Column<string>(type: "nvarchar(450)", nullable: true),
                    UniqueRecordId = table.Column<string>(type: "NVARCHAR COLLATE SQL_Latin1_General_CP1_CS_AS(4000)", nullable: true),
                    IsDeleted = table.Column<bool>(type: "bit", nullable: false)
                },
                constraints: table =>
                {
                    table.PrimaryKey("PK_ObjectIds", x => x.Id);
                });
```

But the generated code is not valid.
instead of this line:

```csharp
UniqueRecordId = table.Column<string>(type: "NVARCHAR COLLATE SQL_Latin1_General_CP1_CS_AS(4000)", nullable: true),
```

Must changed to:

```csharp
UniqueRecordId = table.Column<string>(type: "NVARCHAR(4000) COLLATE SQL_Latin1_General_CP1_CS_AS", nullable: true),
```

To works for Update-Database.

EF Core version:
Database provider: (e.g. Microsoft.EntityFrameworkCore.SqlServer)
Target framework: (e.g. .NET 5.0.2)
Operating system:
IDE: (e.g. Visual Studio 2019 16.9.1)

**Comment by roji** [comment]
Rather than specifying the collation as part of the column type, have you tried using [the dedicated EF API for specifying the collation separately](https://docs.microsoft.com/en-us/ef/core/miscellaneous/collations-and-case-sensitivity#column-collation)?

**Comment by Ali-YousefiTelori** [comment]
@roji Thank you this way working, but ew problem I got and I think its a bug.After add migration I see this code:

```csharp

            migrationBuilder.AlterDatabase(,
                oldCollation: "SQL_Latin1_General_CP1_CI_AS");
```

I changed it to:

```csharp

            migrationBuilder.AlterDatabase(null,
                oldCollation: "SQL_Latin1_General_CP1_CI_AS");
```

And It's fixed now.

**Comment by roji** [comment]
@Ali-YousefiTelori that looks like #23794, which was fixed in 5.0.2. It's always recommended to use the latest patch version.
